app/validators/phone_number_validator.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class PhoneNumberValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    return if value.blank? || value.delete("-+ ()").match(/\A\d{8,15}\z/)
```

Comment:
[I assume you're calling this??](https://ruby-doc.org/core-3.1.2/String.html#method-i-delete)
value.delete("-+ ()")